galata/test/realtime-collaboration/notebook.test.ts

```diff
@@ -0,0 +1,275 @@
+
+test.describe('Initialization', () => {
+  const pathUntitled = 'Untitled.ipynb';
+  const exampleNotebook = 'OutputExamples.ipynb';
+  let guestPage: IJupyterLabPageFixture;
+
+  test.beforeEach(
+    async ({
+      request,
+      appPath,
+      autoGoto,
+      baseURL,
+      browser,
+      mockSettings,
+      mockState,
+      sessions,
+      terminals,
+      tmpPath,
+      waitForApplication
+    }) => {
+      const contents = galata.newContentsHelper(request);
+      await contents.uploadFile(
+        path.resolve(
+          __dirname,
+          `../../../examples/notebooks/${exampleNotebook}`
+        ),
+        `${tmpPath}/${exampleNotebook}`
+      );
+
+      // Create a new client
+      const user: Partial<User.IUser> = {
+        identity: {
+          username: 'jovyan_2',
+          name: 'jovyan_2',
+          display_name: 'jovyan_2',
+          initials: 'JP',
+          color: 'var(--jp-collaborator-color2)'
+        }
+      };
+      guestPage = await galata.newPage(
+        appPath,
+        autoGoto,
+        baseURL!,
+        browser,
+        mockSettings,
+        mockState,
+        user,
+        sessions,
+        terminals,
+        tmpPath,
+        waitForApplication
+      );
+    }
+  );
+
+  test.afterEach(async ({ request, tmpPath }) => {
+    const contents = galata.newContentsHelper(request);
+    await contents.deleteFile(`${tmpPath}/${exampleNotebook}`);
+    // Make sure to close the page to remove the client
+    // from the awareness
+    await guestPage.close();
+  });
+
+  test('Create a notebook', async ({ page }) => {
+    // Renaming does not work
+    await page.notebook.createNew();
+    await guestPage.notebook.open(pathUntitled);
+
+    // wait for kernel to be idle
+    await guestPage.waitForTimeout(1000);
+    expect(await guestPage.screenshot()).toMatchSnapshot();
+
+    // Make sure to close the page to remove the client
+    // from the awareness
+    await guestPage.close();
+  });
+
+  test('Open a notebook', async ({ page }) => {
+    await page.filebrowser.refresh();
+    await page.notebook.open(exampleNotebook);
+    await page.waitForCondition(
+      () => page.isVisible('text=Select Kernel'),
+      1000
+    );
+    if (await page.isVisible('text=Select Kernel')) {
+      await page.keyboard.press('Enter');
+    }
+
+    // wait for kernel to be idle
+    await page.waitForTimeout(2000);
+    expect(await page.screenshot()).toMatchSnapshot();
+
+    await guestPage.filebrowser.refresh();
+    await guestPage.notebook.open(exampleNotebook);
+    if (await guestPage.isVisible('text=Select Kernel')) {
+      await guestPage.keyboard.press('Enter');
+    }
+
+    // wait for kernel to be idle
+    await guestPage.waitForTimeout(1000);
+    expect(await guestPage.screenshot()).toMatchSnapshot();
+
+    // Make sure to close the page to remove the client
+    // from the awareness
+    await guestPage.close();
+  });
+});
+
+test.describe('With 2 clients', () => {
```

Comment:
We could probably go with 10 directly as it 2 or 10 clients are handled the same.